core/templates/pages/exploration-editor-page/translation-tab/state-translation-editor/state-translation-editor.component.spec.ts

```diff
@@ -0,0 +1,395 @@
+      $q = $injector.get('$q');
+      $rootScope = $injector.get('$rootScope');
+      $uibModal = $injector.get('$uibModal');
+      editabilityService = $injector.get('EditabilityService');
+      explorationStatesService = $injector.get('ExplorationStatesService');
+      translationLanguageService = $injector.get('TranslationLanguageService');
+      translationTabActiveContentIdService = $injector.get(
+        'TranslationTabActiveContentIdService');
+
+      spyOn(stateEditorService, 'getActiveStateName').and.returnValue(
+        stateName);
+      spyOn(editabilityService, 'isEditable').and.returnValue(true);
+      stateObj = stateObjectFactory.createFromBackendDict(stateName, state);
+      spyOn(explorationStatesService, 'getState').and.returnValue(stateObj);
+      spyOn(explorationStatesService, 'saveWrittenTranslations').and.callFake(
+        () => {});
+
+      stateWrittenTranslationsService.init(stateName, {
+        hasWrittenTranslation: () => true,
+        getWrittenTranslation: () => (
+          writtenTranslationObjectFactory.createFromBackendDict({
+            html: 'This is a html',
+            needs_update: true
+          })
+        ),
+        updateWrittenTranslationHtml: () => {}
+      });
+
+      $scope = $rootScope.$new();
+      var ctrl = $componentController('stateTranslationEditor', {
+        $scope: $scope,
+        StateEditorService: stateEditorService,
+        StateWrittenTranslationsService: stateWrittenTranslationsService,
+        WrittenTranslationObjectFactory: writtenTranslationObjectFactory
+      });
+      ctrl.$onInit();
+    }));
+
+    it('should evaluate $scope properties after controller initialization',
+      function() {
+        expect($scope.translationEditorIsOpen).toBe(false);
+        expect($scope.activeWrittenTranslation).toEqual(
+          writtenTranslationObjectFactory.createFromBackendDict({
+            html: 'This is a html',
+            needs_update: true
+          }));
+      });
+
+    it('should update state\'s recorded voiceovers after broadcasting' +
+      ' externalSave event when closing modal', function() {
+      $scope.openTranslationEditor();
+      expect($scope.translationEditorIsOpen).toBe(true);
+      stateWrittenTranslationsService.displayed = {
+        hasWrittenTranslation: () => true,
+        getWrittenTranslation: () => (
+          writtenTranslationObjectFactory.createFromBackendDict({
+            html: 'This is a second html',
+            needs_update: true
+          })
+        )
+      };
+      spyOn(translationTabActiveContentIdService, 'getActiveContentId').and
+        .returnValue('content_1');
+      spyOn(translationLanguageService, 'getActiveLanguageCode').and
+        .returnValue('es');
+      spyOn($uibModal, 'open');
+
+      $rootScope.$broadcast('externalSave');
+
+      expect($uibModal.open).not.toHaveBeenCalled();
+    });
+
+    it('should update state\'s recorded voiceovers after broadcasting' +
```

Comment:
this same test seems to be repeated multiple times?

Reply:
Done.